### PR TITLE
luarocks: 3.9.1 -> 3.9.2

### PR DESCRIPTION
**Pull request by jtbx**
## Description of changes

Changes: https://github.com/luarocks/luarocks/blob/master/CHANGELOG.md#whats-new-in-luarocks-392
Diff: https://github.com/luarocks/luarocks/compare/v3.9.1...v3.9.2

Also migrated to pkgs/by-name.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
as per pkgs/by-name check

>  pkgs.luarocks: This attribute is manually defined (most likely in pkgs/top-level/all-packages.nix), which is only allowed if the definition is of the form `pkgs.callPackage pkgs/by-name/lu/luarocks/package.nix { ... }` with a non-empty second argument.

this is probably due to this in all-packages.nix
```
  luarocks = luaPackages.luarocks;
  luarocks-nix = luaPackages.luarocks-nix;
```

maybe @infinisil can suggest a solution ?

**Comment by AndersonTorres** [comment]
> maybe @infinisil can suggest a solution ?

I believe it will require way more advanced features not yet present on the current by-name structure.

**Comment by kirillrdy** [comment]
if that's the case, then we should move files back

**Comment by infinisil** [comment]
The error [should be improved](https://github.com/NixOS/nixpkgs/issues/286396), but indeed `pkgs/by-name` can't yet be used for non-top-level packages.

**Comment by jtbx** [comment]
Okay, in that case I'll move the luarocks package back where it was before

**Review by teto — APPROVED** [review]

File: pkgs/development/tools/misc/luarocks/default.nix [hunk 1]
```diff
@@ -19,23 +19,17 @@
...
   src = fetchFromGitHub {
     owner = "luarocks";
     repo = "luarocks";
     rev = "v${finalAttrs.version}";
-    sha256 = "sha256-G6HDap3pspeQtGDBq+ukN7kftDaT/CozMVdYM60F6HI=";
+    hash = "sha256-D5jH7QuPayDNskIhiPeqKzI9m33MjRjQ8ReghFIUlPo=";
   };
 
   patches = [
     ./darwin-3.7.0.patch
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/luarocks/default.nix:32]: What does this patch do?

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**